R/find.tol.R

```diff
@@ -1,29 +1,33 @@
-find.tol <-
-function(a, uppermost=1e-4, aver.bin.size=4000, min.bins=50, max.bins=200, do.plot=TRUE)
-{
-    a<-a[order(a)]
-    l<-length(a)
-    da<-(a[2:l]-a[1:(l-1)])/((a[2:l]+a[1:(l-1)])/2)
-    da<-da[da < uppermost]
-    n<-min(max.bins, max(round(length(da)/aver.bin.size), min.bins))
-    des<-density(da,kernel="gaussian",n=n, bw=uppermost/n*2,from=0)
-    y<-des$y[des$x>0]
-    x<-des$x[des$x>0]
+find.tol <- function(mz_values,
+                     mz_max_diff = 1e-4,
+                     aver.bin.size = 4000,
+                     min.bins = 50,
+                     max.bins = 200,
+                     do.plot = TRUE) {
+    mz_values <- mz_values[order(mz_values)]
+    l <- length(mz_values)
+    da <- (mz_values[2:l] - mz_values[1:(l - 1)]) / ((mz_values[2:l] + mz_values[1:(l - 1)]) / 2)
+    da <- da[da < mz_max_diff]
+    n <- min(max.bins, max(round(length(da) / aver.bin.size), min.bins))
+    des <- density(da, kernel = "gaussian", n = n, bw = mz_max_diff / n * 2, from = 0)
+    y <- des$y[des$x > 0]
+    x <- des$x[des$x > 0]
+    
+    to.use <- da[da > max(da) / 4] - max(da) / 4
+    this.rate <- MASS::fitdistr(to.use, "exponential")$estimate
+    exp.y <- dexp(x, rate = this.rate)
+    exp.y <- exp.y * sum(y[x > max(da) / 4]) / sum(exp.y[x > max(da) / 4])
+    
+    yy <- cumsum(y > 1.5 * exp.y)
+    yi <- seq_along(yy)
+    sel <- min(which(yy < yi)) - 1
     
-    to.use<-da[da>max(da)/4]-max(da)/4
-    this.rate<-MASS::fitdistr(to.use, "exponential")$estimate
-    exp.y<-dexp(x,rate=this.rate)
-    exp.y<-exp.y*sum(y[x>max(da)/4])/sum(exp.y[x>max(da)/4])
-
-    yy<-cumsum(y>1.5*exp.y)
-    yi<-seq_along(yy)
-    sel<-min(which(yy<yi))-1
-
     if (do.plot) {
-        plot(x,y, xlab="Delta",ylab="Density",main="find m/z tolerance",cex=.25)
-        lines(x,exp.y,col="red")
-        abline(v=x[sel],col="blue")
+        plot(x, y, xlab = "Delta", ylab = "Density",
+             main = "find m/z tolerance", cex = .25)
+        lines(x, exp.y, col = "red")
+        abline(v = x[sel], col = "blue")
```

Comment:
could be moved to plotting section

Reply:
Extracted in [ee5a295](https://github.com/RECETOX/recetox-aplcms/pull/63/commits/ee5a2950c9de8fad8d302a8321fc5c21cb0ca213).